addons/sale/models/account_reconciliation_widget.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, models
+
+
+class AccountReconciliation(models.AbstractModel):
+    _inherit = 'account.reconciliation.widget'
+
+
+    def _get_sales_order(self, st_line):
+        # Search if we can find a sale order line that match the statement reference
+        sols = self.env['sale.order.line'].search([('name', 'ilike', st_line.get('name'))])
```

Comment:
sols = self.env['sale.order.line'].search([('name', 'ilike', st_line.get('name')), ('invoice_status', '=', 'to invoice'), ('company_id', '=', some_company)])

and that's it.. what dayathink?